modules/packages/LinearAlgebra.chpl

```diff
@@ -444,16 +444,19 @@ proc eye(Dom: domain(2), type eltType=real) {
 proc setDiag (ref X: [?D] ?eltType, in k: int = 0, val: eltType = 0)
```

Comment:
I think we should check that:

1. `X.rank == 2`
2. `isSquare(X)`

and give a user-friendly error message if not (see other functions for examples of this).